uaclient/cli.py

```diff
@@ -834,21 +845,74 @@ def action_config_set(args, *, cfg, **kwargs):
         livepatch_status, _ = entitlement.application_status()
         if livepatch_status == ua_status.ApplicationStatus.ENABLED:
             configure_livepatch_proxy(**kwargs)
-    elif set_key in ("apt_http_proxy", "apt_https_proxy"):
+    elif set_key in cfg.ua_scoped_proxy_options:
+        protocol_type = set_key.split("_")[2]
+        if protocol_type == "http":
+            validate_url = util.PROXY_VALIDATION_APT_HTTP_URL
+        else:
+            validate_url = util.PROXY_VALIDATION_APT_HTTPS_URL
+        util.validate_proxy(protocol_type, set_value, validate_url)
+        unset_current = bool(
+            cfg.global_apt_http_proxy or cfg.global_apt_https_proxy
+        )
+        if unset_current:
+            print(
+                messages.WARNING_APT_PROXY_OVERWRITE.format(
+                    current_proxy="ua scoped apt", previous_proxy="global apt"
+                )
+            )
+        kwargs = {
+            "http_proxy": cfg.ua_apt_http_proxy,
+            "https_proxy": cfg.ua_apt_https_proxy,
+            "proxy_scope": AptProxyScope.UACLIENT,
+        }
+        if "https" in set_key:
+            kwargs["https_proxy"] = set_value
+        else:
+            kwargs["http_proxy"] = set_value
+
+        setup_apt_proxy(**kwargs)
```

Comment:
I think this part of the code looks really similar to the one we have for the `global` scenario. I think we can create a function for it